### DISCUSSION TITLE
Should CanExecuteDependsOnAttribute exist to replace NotifyCanExecuteChangedForAttribute?

### DISCUSSION THREAD

**Discussion by suugbut** [CommunityToolkit/dotnet#885] [General]
In my opinion `ObservableProperty` (as a publisher) should not know anything about `ICommand` (as a subscriber). So the following
```cs
public partial class PersonViewModel : ObservableObject
{
    [ObservableProperty]
    [NotifyCanExecuteChangedFor(nameof(SaveCommand))]
    private string name;

    [RelayCommand(CanExecute = nameof(CanSave))]
    private void Save() { }
    private bool CanSave() => !string.IsNullOrWhiteSpace(Name);
}
```
is more elegant if it can be written as follows.
```cs
public partial class PersonVM : ObservableObject
{
    [ObservableProperty]
    private string name;

    [RelayCommand(CanExecute = nameof(CanSave))]
    private void Save() { }
    
    [CanExecuteDependsOn(nameof(Name))] // <======= the requested attribute.
    private bool CanSave() => !string.IsNullOrWhiteSpace(Name);
}
```
With `CanExecuteDependsOnAttribute` we don't need to locate and decorate `ObservableProperty` whenever commands depend on it. It is more convenient if we place all setup related to a command in one place rather than two places spanned across multiple lines. 

Any comments are welcome.

**Comment by pictos** [comment]
Hey @Sergio0694, I believe this discussion should be on MVVM repo, can you move it?
